flake.nix

```diff
@@ -39,6 +39,8 @@
       supportedSystems = import ./supported-systems.nix;
       defaultSystem = head supportedSystems;
 
+      inputMap = { "https://input-output-hk.github.io/cardano-haskell-packages" = CHaP; };
```

Comment:
Maybe we should be more explicit how this works?

We want to ensure that we do get a pinned version of CHaP (as we want to minimise the risk of someone
impersonating GitHub, and subtly giving us some modified CHaP). Thus we use `nix` to lock the CHaP reference,
and then need to pass that locked reference to haskell.nix

1. We reference iog/cardano-haskell-packags in the flake and give it the name `CHaP`.
2. We need to inform haskell.nix about this custom hackage. For this we tell haskell.nix to map the `repository` to the location of `CHaP` from the flake. We record this mapping in the `inputMap`
3. Finally we need pass that `inputMap` to haskell.nix, so it knows where to find the custom hackage.